packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -831,6 +842,22 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin {
     return true;
   }
 
+  /// Complete the lifecycle for a route that has been popped off the navigator.
+  ///
+  /// When the navigator pops a route, the navigator retains a reference to the
+  /// route in order to call [Route.dispose] if the navigator itself is removed
+  /// from the tree. When the route is finished with any exit animation, the
+  /// route should call this function to complete its lifecycle (e.g., to
+  /// receive a call to [Route.dispose]).
+  ///
+  /// The given `route` must have already received a call to [Route.didPop].
+  /// Maybe be called re-entrantly from [Route.didPop] if [Route.didPop] will
```

Comment:
Typo: Maybe be => May be.

I'm having some trouble understanding the explanation about how finalizeRoute() can be reentered.